MapboxNavigation/TopBannerViewController.swift

```diff
@@ -0,0 +1,433 @@
+        instructionsBannerView.update(for: instruction)
+        lanesView.update(for: instruction)
+        nextBannerView.update(for: instruction)
+    }
+    
+    public func navigationService(_ service: NavigationService, willRerouteFrom location: CLLocation) {
+        let title = NSLocalizedString("REROUTING", bundle: .mapboxNavigation, value: "Rerouting…", comment: "Indicates that rerouting is in progress")
+        lanesView.hide()
+        statusView.show(title, showSpinner: true)
+    }
+    
+    public func navigationService(_ service: NavigationService, didRerouteAlong route: Route, at location: CLLocation?, proactive: Bool) {
+        instructionsBannerView.updateDistance(for: service.routeProgress.currentLegProgress.currentStepProgress)
+        
+        dismissStepsTable()
+        if service.simulationMode == .always {
+            let localized = String.Localized.simulationStatus(speed: Int(service.simulationSpeedMultiplier))
+            statusView.showStatus(title: localized, for: .infinity, animated: true, interactive: true)
+        } else {
+            statusView.hide(delay: 2, animated: true)
+        }
+        
+        if (proactive) {
+            let title = NSLocalizedString("FASTER_ROUTE_FOUND", bundle: .mapboxNavigation, value: "Faster Route Found", comment: "Indicates a faster route was found")
+            statusView.showStatus(title: title, withSpinner: true, for: 3)
+        }
+    }
+    
+    public func navigationService(_ service: NavigationService, willBeginSimulating progress: RouteProgress, becauseOf reason: SimulationIntent) {
+        guard reason == .manual else { return }
+        let localized = String.Localized.simulationStatus(speed: 1)
+        statusView.show(localized, showSpinner: false, interactive: true)
+    }
+    
+    public func navigationService(_ service: NavigationService, willEndSimulating progress: RouteProgress, becauseOf reason: SimulationIntent) {
+        guard reason == .manual else { return }
+        statusView.hide(delay: 0, animated: true)
+    }
+    
+    public func navigationViewController(_ controller: NavigationViewController, didRecenterAt location: CLLocation) {
+        stopPreviewing()
+    }
+    
+    private func embed(_ child: UIViewController, in container: UIView, constrainedBy constraints: ((UIViewController, UIViewController) -> [NSLayoutConstraint])? = nil) {
+        child.willMove(toParent: self)
+        addChild(child)
+        container.addSubview(child.view)
+        if let childConstraints: [NSLayoutConstraint] = constraints?(self, child) {
+            view.addConstraints(childConstraints)
+        }
+        child.didMove(toParent: self)
+    }
+}
+
+// MARK: InstructionsBannerViewDelegate Conformance
+extension TopBannerViewController: InstructionsBannerViewDelegate {
+    public func didTapInstructionsBanner(_ sender: BaseInstructionsBannerView) {
+        if isDisplayingSteps {
```

Comment:
Unsatisfiable constraints warning triggers when toggling the steps in rapid succession.
Prevent interrupting ongoing interaction if we can't have adaptive constraints.
Adaptive interruptable constraints animations via UIViewPropertyAnimator requires iOS 10+ :/ 

<details>

```
	Probably at least one of the constraints in the following list is one you don't want. 
	Try this: 
		(1) look at each constraint and try to figure out which you don't expect; 
		(2) find the code that added the unwanted constraint or constraints and fix it. 
(
    "<NSLayoutConstraint:0x6000013b0190 V:[UIStackView:0x7fd13c74fc30]-(0)-[UIView:0x7fd13c74fa50]   (active)>",
    "<NSLayoutConstraint:0x6000013d8cd0 MBDismissButton:0x7fd13d303b70'Close'.height == 54   (active)>",
    "<NSLayoutConstraint:0x6000013db890 MBDismissButton:0x7fd13d303b70'Close'.bottom == UILayoutGuide:0x600000af32c0'UIViewSafeAreaLayoutGuide'.bottom   (active)>",
    "<NSLayoutConstraint:0x6000013d9360 V:|-(0)-[UITableView:0x7fd13beff400]   (active, names: '|':UIView:0x7fd139f91320 )>",
    "<NSLayoutConstraint:0x6000013db610 UITableView:0x7fd13beff400.bottom == MBDismissButton:0x7fd13d303b70'Close'.top   (active)>",
    "<NSLayoutConstraint:0x60000104b6b0 V:|-(0)-[UIView:0x7fd139f91320]   (active, names: '|':UIView:0x7fd13c74fa50 )>",
    "<NSLayoutConstraint:0x60000104a170 UIView:0x7fd139f91320.bottom == UIView:0x7fd13c74fa50.bottom   (active)>",
    "<NSLayoutConstraint:0x60000105fca0 V:[UIStackView:0x7fd13c74fc30]-(0)-|   (active, names: '|':UIView:0x7fd13c74ed10 )>",
    "<NSLayoutConstraint:0x6000013b0460 V:[UIView:0x7fd13c74fa50]-(0)-|   (active, names: '|':UIView:0x7fd13c74ed10 )>",
    "<NSLayoutConstraint:0x6000013d8d70 'UIViewSafeAreaLayoutGuide-bottom' V:[UILayoutGuide:0x600000af32c0'UIViewSafeAreaLayoutGuide']-(0)-|   (active, names: '|':UIView:0x7fd139f91320 )>"
)

Will attempt to recover by breaking constraint 
<NSLayoutConstraint:0x6000013d8cd0 MBDismissButton:0x7fd13d303b70'Close'.height == 54   (active)>
```

</details>